scala-package/pom.xml

```diff
@@ -117,31 +90,66 @@
     </profile>
 
     <profile>
-      <id>scala-2.12</id>
+      <id>osx-x86_64</id>
+      <activation>
+        <os>
+          <family>mac</family>
+        </os>
+      </activation>
+      <properties>
+        <platform>osx-x86_64</platform>
+        <libtype>jnilib</libtype>
+        <flavor>cpu</flavor>
+      </properties>
+    </profile>
+    <profile>
+      <id>linux-x86_64</id>
+      <activation>
+        <os>
+          <family>unix</family>
+          <name>Linux</name>
+        </os>
+      </activation>
       <properties>
-        <scala.version>2.12.4</scala.version>
-        <scala.binary.version>2.12</scala.binary.version>
+        <platform>linux-x86_64</platform>
+        <libtype>so</libtype>
       </properties>
+
       <build>
         <plugins>
           <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-enforcer-plugin</artifactId>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
+            <version>1.6.0</version>
             <executions>
               <execution>
-                <id>enforce-versions</id>
+                <id>init-build-flavor</id>
+                <phase>initialize</phase>
                 <goals>
-                  <goal>enforce</goal>
+                  <goal>exec</goal>
                 </goals>
                 <configuration>
-                  <rules>
-                    <bannedDependencies>
-                      <excludes combine.children="append">
-                        <exclude>*:*_2.11</exclude>
-                        <exclude>*:*_2.10</exclude>
-                      </excludes>
-                    </bannedDependencies>
-                  </rules>
+                  <executable>bash</executable>
+                  <commandlineArgs>-c 'mkdir -p ${project.build.directory}; if [[ $(ldd ${MXNET_DIR}/lib/libmxnet.so | grep libcuda.so | wc -l) == "0" ]]; then echo flavor=cpu &gt; ${project.build.directory}/flavor.properties; else echo flavor=gpu &gt; ${project.build.directory}/flavor.properties; fi'</commandlineArgs>
```

Comment:
A little concern, if I happened to forget clean my build stuff (I have libcuda on my laptop). Will this cause the maven to consider the project as GPU if I just need to build with CPU?

Reply:
For scala maven point of view, the .so file must pre-exist in the lib folder. It has to be either CPU or GPU version, you can't have both.
Yes, if you have a GPU .so file in lib folder, the output of maven build is GPU version. And it will be published as mxnet-full_2.11-${platform}-gpu, you won't accidentally publish to cpu package.